### PR TITLE
Robin's Feedback

### PR DESCRIPTION
**Pull request by Zaubeerer**

### CODE REVIEW

**Comment by Zaubeerer** [comment]
Hey @lucasiragusa, I just had a first look into your code: Great job! 

We can refactor it a bit and then add streamlit and fastapi on top of it :)

**Comment by Zaubeerer** [comment]
Oh, normally I would have suggested to first have discussions on this PR and create follow-up issues.

We can set the repo up such that you can't merge if conversations are not resolved. 

We will do that Friday.

For now, I suggest to create follow-up issues, according to the conversations I am now creating.

File: src/lib/comparer.py [hunk 1]
```diff
@@ -0,0 +1 @@
+# TODO: what shall this be about?
```
Comment (Zaubeerer) [inline src/lib/comparer.py:1]: Implement comparer

File: src/lib/permit_generator.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import sys
-import os 
+import os
 import pandas as pd
 
+# TODO: install lib as package instead
```
Comment (Zaubeerer) [inline src/lib/permit_generator.py:5]: Set up package that can be installed with `pip install -e .`

File: src/lib/permit_generator.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
 # This will get the path to the parent directory of the current directory (i.e., landingpermit_app)
 sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
Comment (Zaubeerer) [inline src/lib/permit_generator.py:7]: Switch to `pathlib` instead of `os.path`

File: src/lib/permit_generator.py [hunk 4]
```diff
@@ -12,8 +13,8 @@
 from docx.shared import Inches
 
 
-def generate_document(country, ssim_file, airline_name, contact_person, handler):
-    
+def generate_document(country, ssim_file, airline_name, contact_person, handler): # TODO: add type hints?
```
Comment (Zaubeerer) [inline src/lib/permit_generator.py:16]: add type annotations to the code

File: src/lib/permit_generator.py [hunk 5]
```diff
@@ -12,8 +13,8 @@
...
+    # TODO: add docstring
```
Comment (Zaubeerer) [inline src/lib/permit_generator.py:17]: add docstrings to functions, module strings to files and maybe further comments and docs

File: src/models/airport_data_loader.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import pandas as pd
 
+# TODO: move this into the airport class
 airport_data = pd.read_pickle('../data/industry/airport_data.pkl')
```
Comment (Zaubeerer) [inline src/models/airport_data_loader.py:4]: Create `read_from_pickle` method in airport class

File: src/models/country.py [hunk 7]
```diff
@@ -0,0 +1 @@
+# TODO: what shall this do?
```
Comment (Zaubeerer) [inline src/models/country.py:1]: Create country model

File: src/models/country_handler.py [hunk 8]
```diff
@@ -0,0 +1 @@
+# TODO: what shall this do?
```
Comment (Zaubeerer) [inline src/models/country_handler.py:1]: Implement country_handler

File: src/models/flight_handler.py [hunk 9]
```diff
@@ -0,0 +1 @@
+# TODO: what shall this do?
```
Comment (Zaubeerer) [inline src/models/flight_handler.py:1]: Implement flight_handler

File: src/models/seasons_handler.py [hunk 10]
```diff
@@ -10,6 +10,7 @@ def __init__(self, start_date, end_date):
 
     # Helper function to determine the season of a given date
 
+    # TODO: isn't there already a built-in function for this? 
     def determine_season(self, date):
```
Comment (Zaubeerer) [inline src/models/seasons_handler.py:14]: Check if determine_season can be done with built-in functions?

File: src/utils/file_helper.py [hunk 11]
```diff
@@ -0,0 +1 @@
+# TODO: What shall this include?
```
Comment (Zaubeerer) [inline src/utils/file_helper.py:1]: Delete `file_helper.py`?

**Review by Zaubeerer — COMMENTED** [review]
As mentioned above, let's create new issues for the following conversations. Therefore, you can click on the three dots and then "open in new tab" the "Reference in issue".

image

**Comment by lucasiragusa** [comment]
Okay,

Sorry this exposes my lack of knowledge on how to actually use git.

All the comments are clear and I understand how to start addressing them
but I just thought I would merge and make the changes in the merged file.

Keen to understand on Friday how to manage this branching and issues
situation that I never worked with before.

Thanks!
Luca

**Comment by Zaubeerer** [comment]
No problem! That is a totally normal process at the beginning :)

**Review by Zaubeerer — COMMENTED** [review]
intermediate comment

File: src/lib/permit_generator.py (hunk 4 above)
Comment (Zaubeerer) [reply src/lib/permit_generator.py:16]: Please create a follow up issue for this

File: src/models/seasons_handler.py (hunk 10 above)
Comment (Zaubeerer) [reply src/models/seasons_handler.py:14]: Please create a follow up issue for this

**Pull request merged**